examples/random-login/main.go

```diff
@@ -20,32 +19,37 @@ import (
 )
 
 func main() {
-	err := p.RunProvider(context.Background(), "random-login", "0.1.0", provider())
+	provider, err := provider()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error: %s", err.Error())
+		os.Exit(1)
+	}
+	err = p.RunProvider(context.Background(), "random-login", "0.1.0", provider)
 	if err != nil {
 		fmt.Fprintf(os.Stderr, "Error: %s", err.Error())
 		os.Exit(1)
 	}
 }
 
-func provider() p.Provider {
-	return infer.Provider(infer.Options{
-		Resources: []infer.InferredResource{infer.Resource[*RandomSalt]()},
-		Components: []infer.InferredComponent{
+func provider() (p.Provider, error) {
+	return infer.NewProviderBuilder().
+		WithResources(
+			infer.Resource[*RandomSalt](),
+		).
+		WithComponents(
 			infer.Component(NewRandomLogin),
 			infer.Component(NewMoreRandomPassword),
-		},
-		Config: infer.Config[Config](),
```

Comment:
@blampe should it be `infer.Config[Config]` or `infer.Config[*Config]`?

Reply:
`infer.Config[Config]` is fine since it doesn't have any pointer methods.